<img src="../images/topcover.jpg" width="1000" height="50">

## Colorectal cancer dataset

In [1]:
# imports relevant modules

import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import confusion_matrix, plot_confusion_matrix, f1_score
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVC

In [2]:
colorectalx = pd.read_csv('../data/hmnist_64_64_L.csv')
colorectalx.drop(columns = 'label', inplace=True)
colorectaly = pd.read_csv('../data/hmnist_64_64_L.csv')

In [3]:
# set up data for modelling svm

X = colorectalx
y = colorectaly['label']

In [4]:
# Check distribution since this is a classification problem

y.value_counts(normalize = True)

7    0.125
3    0.125
6    0.125
2    0.125
5    0.125
1    0.125
8    0.125
4    0.125
Name: label, dtype: float64

In [5]:
# split the data into the training and testing sets

X_train, X_val, y_train, y_val = train_test_split(X,
                                                    y,
                                                    test_size=0.33,
                                                    stratify=y,
                                                    random_state=42
                                                    )

In [6]:
# standard scaler applied

ss = StandardScaler()
ss.fit(X_train)
X_train = ss.transform(X_train)
X_val = ss.transform(X_val)

## Support Vector Machine Model

In [7]:
# Instantiate support vector machine.
svc = SVC()

In [8]:
gs1 = GridSearchCV(estimator=SVC(),
             param_grid={'C': [1, 10], 'kernel': ('linear', 'rbf', 'poly'), 'degree':[2]})
gs1.fit(X_train,y_train);

In [10]:
predictions = gs1.predict(X_val)

In [11]:
gs1.score(X_train, y_train)

0.7937313432835821

In [12]:
gs1.score(X_val, y_val)

0.636969696969697

##### The summary table below shows that Random Forests has the highest score, followed by SVM and CNN for classification of image data into 8 classes of tissue types. However, pathologists and even researchers may place emphasis on 2 particular tissue types, namely, Stroma and Tumour. Therefore images from these 2 classes will be classified in the next notebook. 

**Summary table for Colorectal Cancer Classification models for 8 Tissue type classes:**

| Model| Test Accuracy|Baseline score|
|:---------:|:---:|:--------:|
|  Random Forests |    0.684 |  0.125  |
|KNN| 0.432| 0.125|
|CNN|  0.603 |0.125|
|SVM| 0.637|0.125|